unyt/unit_object.py

```diff
@@ -331,6 +331,10 @@ def __repr__(self):
     def __str__(self):
         if self.expr == sympy_one:
             return "dimensionless"
+        if self.expr.__str__() == "degC":
```

Comment:
Sorry that I missed this earlier, can you inline the call to `expr.__str__()`? Printing and string generation can unfortunately sometimes be in performance-sensitive code paths so if we can avoid calling that function three times we should.